deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sResourceDispatcher.java

```diff
@@ -165,105 +184,112 @@ public void undeployApp(String appName) throws HyscaleException {
      * @throws HyscaleException if failed to delete any resource
      */
     public void undeployService(String appName, String serviceName) throws HyscaleException {
-        logger.info("Undeploy initiated for service - {}",serviceName);
+        logger.info("Undeploy initiated for service - {}", serviceName);
         if (StringUtils.isBlank(appName)) {
             logger.error("No applicaton found for undeployment");
             throw new HyscaleException(DeployerErrorCodes.APPLICATION_REQUIRED);
         }
-        PodParentUtil podParentUtil = new PodParentUtil(apiClient,namespace);
+        PodParentUtil podParentUtil = new PodParentUtil(apiClient, namespace);
         PodParent podParent = podParentUtil.getPodParentForService(serviceName);
         List<CustomResourceKind> appliedKindsList = podParentUtil.getAppliedKindsList(podParent);
-        String selector = ResourceSelectorUtil.getServiceSelector(appName,serviceName);
-        deleteResources(selector,appliedKindsList);
+        String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+        deleteResources(selector, appliedKindsList);
     }
 
-    private List<String> buildAppliedKindsAnnotation(MultiValueMap<String, CustomObject> kindVsCustomObject){
-        if(kindVsCustomObject == null || kindVsCustomObject.isEmpty()){
+/*
+    private List<String> buildAppliedKindsAnnotation(MultiValueMap<String, CustomObject> kindVsCustomObject) {
+        if (kindVsCustomObject == null || kindVsCustomObject.isEmpty()) {
             return Collections.emptyList();
         }
         List<String> appliedKinds = new ArrayList<>();
-        kindVsCustomObject.forEach((kind,customObjects)->{
-            String kindVsApiVersion = kind+":"+customObjects.get(0).getApiVersion();
+        kindVsCustomObject.forEach((kind, customObjects) -> {
+            String kindVsApiVersion = kind + ":" + customObjects.get(0).getApiVersion();
             appliedKinds.add(kindVsApiVersion);
         });
         return appliedKinds;
     }
+*/
+
+    private String buildAppliedKindAnnotation(CustomObject customObject){
+        return customObject.getKind() + ":" + customObject.getApiVersion();
+    }
 
-    private void applyCustomResources(MultiValueMap<String, CustomObject> kindVsCustomObject){
-        if(kindVsCustomObject != null && !kindVsCustomObject.isEmpty()){
-            kindVsCustomObject.forEach((kind,customObjects)->{
-                for(CustomObject object : customObjects){
+    private void applyCustomResources(List<CustomObject> customObjectList) {
+//        if (kindVsCustomObject != null && !kindVsCustomObject.isEmpty()) {
+//            kindVsCustomObject.forEach((kind, customObjects) -> {
+                for (CustomObject object : customObjectList) {
```

Comment:
We can use stream for each here
No need to have a null check for genericK8sClient as we only are initializing it